tools/pkg/engine_build_configs/test/build_config_runner_test.dart

```diff
@@ -834,7 +834,7 @@ void main() {
             ..createSync(recursive: true);
       final io.File file =
           io.File(path.join(hostDebug.path, 'compile_commands.json'));
-      file.writeAsString(
+      file.writeAsStringSync(
```

Comment:
I'd expect that an unawaited futures lint would have caught this

Reply:
The Dart analyzer's `unawaited_futures` check does warn about this.

But that check was disabled due to false positives (see https://github.com/flutter/engine/blob/main/analysis_options.yaml#L197)